src/state.rs

```diff
@@ -456,11 +477,31 @@ impl DistributedCtx<LunaticEnvironment> for DefaultProcessState {
             wasi_stderr: None,
             initialized: false,
             registry: Default::default(), // TODO move registry into env?
+            plugins: Arc::new(vec![]),
+            plugin_state: HashMap::new(),
```

Comment:
Should plugins and plugin state be empty in `new_dist_state`?

Reply:
Probably not (I'm not sure :smile: ), when reading through #69 the original idea was to have different plugins for each Environment.

> Testing plugins or comparing performance between competing implementations becomes trivial, just spawn different environments, attach different plugins to them and run the same code. 

When spawning via distributed API we use Environment to crate state.

If I understand this correctly calling a function from a dynamic library will fail, in this case when spawning a process on a different node, since plugin is not linked?